src/JLSOFile.jl

```diff
@@ -56,10 +59,11 @@ function JLSOFile(
         Pkg.TOML.parse(project_toml),
         Pkg.TOML.parse(manifest_toml),
         Dict{Symbol, Vector{UInt8}}(),
+        ReentrantLock()
     )
 
     for (key, val) in data
-        jlso[key] = val
+        @spawn jlso[key] = val
     end
```

Comment:
Need to make sure we don't return until all these tasks have completed.
I think a `@sync` on the `for` might do it?

Reply:
Yes, oversight!